generators/server/templates/quarkus/src/main/java/package/web/rest/AccountResource.java.ejs

```diff
@@ -73,6 +96,119 @@ public class AccountResource {
         this.userService = userService;
     }
 
+    /**
+     * {@code GET /account} : get the current user.
+     *
+     * @return the current user.
+     * @throws RuntimeException {@code 500 (Internal Server Error)} if the user couldn't be returned.
+     */
+    @GET
+    @Path("/account")
+    @Authenticated
+    public UserDTO getAccount(@Context SecurityContext ctx) {
+        return userService
+            .getUserWithAuthoritiesByLogin(ctx.getUserPrincipal().getName())
+            .map(UserDTO::new)
+            .orElseThrow(() -> new AccountResourceException("User could not be found"));
+    }
+<%_ } _%>
+<%_ if (authenticationType === 'oauth2') { _%>
+    /**
+     * {@code GET /account} : get the current user.
+     *
+     * @return the current user.
+     * @throws RuntimeException {@code 500 (Internal Server Error)} if the user couldn't be returned.
+     */
+    @GET
+    @Path("/account")
+    @Authenticated
+    public UserVM getAccount(@Context SecurityContext ctx) {
+        var principal = ctx.getUserPrincipal();
+        if (principal instanceof JsonWebToken) {
+            return getUserFromAuthentication((JsonWebToken) principal);
+        } else {
+            throw new AccountResourceException("User could not be found");
+        }
+    }
+
+    private static UserVM getUserFromAuthentication(JsonWebToken jwt) {
+        UserVM user = new UserVM();
+        // handle resource server JWT, where sub claim is email and uid is ID
+        if (jwt.getClaim("uid") != null) {
```

Comment:
This code seems to be the same than in the UserService. Could we have a common method for both?

Reply:
Actually I needed to remove code from the `UserService` when oauth2 is choosen I do not generate UserService and everything happens in `AccountResource`